### PR TITLE
Route to name most recent site entry for an adopted site

### PR DESCRIPTION
**Pull request by Tvcz**
Created a route for renaming the most recent site entry that exists for a given site id. The site must be adopted by the calling user.

Route accessible via:
`POST api/v1/protected/sites/:site_id/name_entry`

Also created a LinkedResourceDoesNotExist exception in order to give more information in the case of a resource not existing that on its own does not have a meaningful id. The exception responds with a `400 BAD REQUEST` status. An example of the error message is shown below:

Screenshot 2022-03-20 144916

[ClickUp Ticket](https://app.clickup.com/t/22gq4y3)

### CODE REVIEW

**Review by rymaju — APPROVED** [review]
LGTM 🛳⛴🚢🏴‍☠️🏴‍☠️🏴‍☠️🚀🚀🚀🛸

File: api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java [hunk 1]
```diff
@@ -46,4 +47,7 @@ void recordStewardship(
 
   /** Removes the given stewardship activity */
   void deleteStewardship(JWTData userData, int activityId);
+
+  /** Removes the record in the adopted sites table linking the site to its current adopter */
```
Comment (SofieDunt) [inline api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java:51]: ❓ Is this correct?

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 2]
```diff
@@ -374,4 +378,28 @@ public void deleteStewardship(JWTData userData, int activityId) {
...
+      throw new AuthException("User is not the site's adopter.");
+    }
+
+    SiteEntriesRecord siteEntry = db.selectFrom(SITE_ENTRIES)
+        .where(SITE_ENTRIES.SITE_ID.eq(siteId))
+        .orderBy(SITE_ENTRIES.UPDATED_AT)
+        .limit(1)
+        .fetchOne();
+
+    if (siteEntry == null) {
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:394]: 👍 Good check

File: api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java (hunk 1 above)
Comment (Tvcz) [reply api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java]: Nope, thanks. Should be fixed now. :+1:

**Pull request merged**